avr-hal-generic/src/i2c.rs

```diff
@@ -570,5 +570,16 @@ macro_rules! impl_twi_i2c {
                 }
             }
         }
+
+        impl<CLOCK, M> [<$I2c Slave>]<CLOCK, M>
+        where
+            CLOCK: $crate::clock::Clock,
+        {
+            pub fn start(&mut self) {
+                // this does not compile
+                // self.p.$twar.write(|w| unsafe {w.bits(self.address)});
```

Comment:
let's go through the line step by step:

1. `p` is the peripheral.  It's type is passed in from the macro-invocation and e.g. for `ATmega328P` will end up being [`avr_device::atmega328p::TWI`](https://docs.rs/avr-device/0.2.3/avr_device/atmega328p/struct.TWI.html).  This type "derefs" to [`&avr_device::atmega328p::twi::RegisterBlock`](https://docs.rs/avr-device/0.2.3/avr_device/atmega328p/twi/struct.RegisterBlock.html) which is where fields for all the registers are found.
2. `$twsr` is a macro token that the macro-invocation gives a proper register name for.  This exists because the registers might have different names on different MCUs.  For `ATmega328P`, it will just be `twsr`.  The same story with the `$twps` field name for a bit-field in the register.  The line will end up being expanded to
   ```rust
   p.twsr.write(|w| w.twps().prescaler_1()); 
   ```
   for `ATmega328P`.
3. The `.write()` method is part of the `svd2rust` peripheral API.  Read more about it in [its documentation](https://docs.rs/svd2rust/0.17.0/svd2rust/#read--modify--write-api).  Essentially, this line says: Write the whole register to its default values _and_ set the prescaler bits to the bit-pattern for `PRESCALER1`.
4. `w.$twps()` means "take the TWPS field in this register"
5. and `$twps().prescaler_1()` means "write the bit-pattern for `PRESCALER1` to it".  This is also part of the autogenerated API ([link 1](https://docs.rs/avr-device/0.2.3/avr_device/atmega328p/twi/twsr/struct.W.html), [link 2](https://docs.rs/avr-device/0.2.3/avr_device/atmega328p/twi/twsr/struct.TWPS_W.html)).

Hope this helps ...